src/btree/bt_sync_obsolete.c

```diff
@@ -109,46 +164,29 @@ __sync_obsolete_inmem_evict_or_mark_dirty(WT_SESSION_IMPL *session, WT_REF *ref)
-         * load.
+         * Dirty the page with an obsolete time window to let the page reconciliation remove all the
+         * obsolete time window information.
          */
-        if (__wt_atomic_load32(&btree->checkpoint_cleanup_obsolete_tw_pages) >=
-          conn->heuristic_controls.checkpoint_cleanup_obsolete_tw_pages_dirty_max)
-            return (0);
-
-        /* Limit the number of btrees that can be cleaned up. */
-        if (__wt_atomic_load32(&btree->eviction_obsolete_tw_pages) == 0 &&
-          __wt_atomic_load32(&btree->checkpoint_cleanup_obsolete_tw_pages) == 0 &&
-          __wt_atomic_load32(&conn->heuristic_controls.obsolete_tw_btree_count) >=
-            conn->heuristic_controls.obsolete_tw_btree_max)
-            return (0);
-
-        if (__wt_txn_newest_visible_all(session, newest_ta.newest_txn,
-              WT_MAX(newest_ta.newest_start_durable_ts, newest_ta.newest_stop_durable_ts))) {
-            /*
-             * Dirty the page with an obsolete time window to let the page reconciliation remove all
-             * the obsolete time window information.
-             */
-            __wt_verbose(session, WT_VERB_CHECKPOINT_CLEANUP,
-              "%p in-memory page %s obsolete time window: time aggregate %s", (void *)ref, tag,
-              __wt_time_aggregate_to_string(&newest_ta, time_string));
+        __wt_verbose(session, WT_VERB_CHECKPOINT_CLEANUP,
+          "%p in-memory page %s obsolete time window: time aggregate %s", (void *)ref, tag,
+          __wt_time_aggregate_to_string(&newest_ta, time_string));
 
-            WT_RET(__wt_page_modify_init(session, ref->page));
-            __wt_page_modify_set(session, ref->page);
+        WT_RET(__wt_page_modify_init(session, ref->page));
+        __wt_page_modify_set(session, ref->page);
 
-            /*
-             * Save that another tree has been processed if that's the first time it gets cleaned
-             * and update the number of pages made dirty for that tree.
-             */
-            if (__wt_atomic_load32(&btree->eviction_obsolete_tw_pages) == 0 &&
-              __wt_atomic_load32(&btree->checkpoint_cleanup_obsolete_tw_pages) == 0)
-                __wt_atomic_addv32(&conn->heuristic_controls.obsolete_tw_btree_count, 1);
+        /*
+         * Save that another tree has been processed if that's the first time it gets cleaned and
+         * update the number of pages made dirty for that tree.
+         */
+        if (__wt_atomic_load32(&btree->eviction_obsolete_tw_pages) == 0 &&
+          __wt_atomic_load32(&btree->checkpoint_cleanup_obsolete_tw_pages) == 0)
+            __wt_atomic_addv32(&conn->heuristic_controls.obsolete_tw_btree_count, 1);
 
-            __wt_atomic_addv32(&btree->checkpoint_cleanup_obsolete_tw_pages, 1);
-            WT_STAT_CONN_DSRC_INCR(session, checkpoint_cleanup_pages_obsolete_tw);
-        }
+        __wt_atomic_addv32(&btree->checkpoint_cleanup_obsolete_tw_pages, 1);
```

Comment:
This will add a full barrier. We can actually use a relaxed version to modify the data.

Reply:
Understood, created WT-13328 to do that work.